### DISCUSSION TITLE
Adding commands to Sliver (help)

### DISCUSSION THREAD

**Discussion by rwhitcroft** [BishopFox/sliver#763] [General]
Hi all,

A colleague and I are interested in contributing to Sliver. We're trying to add a simple command just to be able to understand how everything fits together, but we're hitting walls.

We looked at the pull request for the `mv` command (https://github.com/BishopFox/sliver/pull/638) and tried to model ours based on that, but we're still having issues. The `mv` PR had to modify 19 files in total which is daunting.

Is there any documentation we can look at to get a better idea of how to go about adding a command?

Cheers, and thank you for all the work you've done.

**Comment by rkervella** [comment]
We currently don't have a proper developer documentation at the moment, the last time we sort of documented on that process was in this issue: https://github.com/BishopFox/sliver/issues/415#issuecomment-828643835

**Reply by rkervella** [reply]
Minus the `srcfiles.go` that is not required anymore (and doesn't exists anymore), the procedure is pretty much up to date.

**Reply by rwhitcroft** [reply]
Thanks for the reply. I'll take a look at that.
